src/autogluon/bench/eval/scripts/run_evaluation_openml.py

```diff
@@ -15,20 +15,115 @@
+        ),
+    ] = None,
+    output_suffix: str = typer.Option(
+        "ag_eval",
+        help="Output suffix of the path to save the output files, e.g. '<results_dir>output/openml/<output_suffix>/'.",
+    ),
+    framework_nan_fill: str = typer.Option(
+        None,
+        help="Framework used as the default result to fill missing values for the frameworks in `frameworks`. E.g., if `framework_nan_fill='foo'`, for dataset A of `framework='bar'` has no result (or NaN result), it is replaced by the result of `'foo'` on dataset A.",
+    ),
+    problem_types: Annotated[
+        Optional[List[str]],
+        typer.Option(help="List of problem types to filter results to. E.g. 'problem_type1,problem_type2,..."),
+    ] = None,
+    folds_to_keep: Annotated[
+        Optional[List[int]], typer.Option(help="List of result folds to use. By default folds 0-9 (10-fold) are used.")
+    ] = None,
+    banned_datasets: Annotated[
+        Optional[List[str]],
+        typer.Option(help="List of datasets to skip during evaluation. E.g. 'dataset1,dataset2,..."),
+    ] = None,
+    infer_batch_size: int = typer.Option(
+        None,
+        help="If specified, will replace the `time_infer_s` column with the value in column `pred_time_test_with_transform_batch_size_{infer_batch_size}`. If a given row does not have a value in the infer_batch_size column, the original `time_infer_s` value is used.",
+    ),
+    compute_z_score: bool = typer.Option(
+        True,
+        help="Whether to compute comparison z-scores. The framework being compared against is the first one listed in `frameworks_run`. Only valid when multiple folds are used and `treat_folds_as_datasets=False`.",
+    ),
+    treat_folds_as_datasets: bool = typer.Option(False, help="If True, treat each fold as a separate dataset."),
+    use_tid_as_dataset_name: bool = typer.Option(
+        True,
+        help="If True, replaces dataset human-readable names with unique integer IDs associated with their OpenML task ID.",
+    ),
+    filter_errors: bool = typer.Option(
+        False,
+        help="If True, any dataset missing a result from any framework in `frameworks` will be filtered from all results. Takes priority over `framework_nan_fill`.",
+    ),
+    clean_data: bool = typer.Option(
+        True,
+        help="If True, performs some general data cleanup to avoid inconsistencies in dataset naming compared to task ids.",
+    ),
+) -> Dict[str, pd.DataFrame]:
+    """
+    Generate evaluation results and saves the results to <results_dir>output/openml/<output_suffix>/evaluation_results.json
+
+    Example:
+    agbench evaluate-amlb-results --frameworks_run framework_1 --frameworks_run framework_2 --paths openml_ag_ag_bench_20230707T070230.csv --results-dir-input data/results/input/prepared/openml --no-clean-data
+    """
+    evaluate(
+        frameworks_run=frameworks_run if frameworks_run else None,
```

Comment:
Why `A if A else None`? What is `A` when we would do `else`?

Reply:
This is due to the fact that `typer` treats `List` default to `[]` instead of `None`. So I'm trying to force `[]` to be 'None` in order to align with `evaluate()` signatures.

Reply:
Ok, long-term we may want to figure out a way to make it actually default to None to avoid this hack. There is also an edge-case where the user actually provides `[]` but it is treated as None incorrectly with the current logic.